src/CLICommand/common_context.ts

```diff
@@ -0,0 +1,89 @@
+import { DriveKey } from '../arfsdao';
+import { JWKWallet, Wallet, WalletDAO } from '../wallet_new';
+import { ParameterName } from './parameter';
+import * as fs from 'fs';
+import { JWKInterface } from 'ardrive-core-js';
+import {
+	DriveAddressParameter,
+	DriveKeyParameter,
+	DrivePasswordParameter,
+	SeedPhraseParameter,
+	WalletFileParameter
+} from '../parameter_declarations';
+import Arweave from 'arweave';
+
+/**
+ * @type {CommonContext}
+ * A class representing the context of the parameters
+ */
+export class CommonContext {
+	private readonly walletDao: WalletDAO;
+
+	/**
+	 * @returns {CommonContext}
+	 * @param {any} options The object containing the parameterName: value mapping
+	 * @param {Arweave} arweave The arweave instance
+	 * An immutable instance of CommonContext holding the parsed values of the parameters
+	 */
+	constructor(private readonly options: any, private readonly arweave: Arweave) {
+		this.walletDao = new WalletDAO(this.arweave);
```

Comment:
Lets hand down the walletDao from index.ts so we're using a single source. The ArDrive class will be using the same walletDao

Reply:
Applied this at at 9ce6e5fa3a937ce4c9e489b8ba6a9bc9dee16b37